### ISSUE TITLE
Log Weight Changes to EEPROM

### ISSUE THREAD

**Issue by alexanderkjones** [alexanderkjones/GM_V1#5]
**Story**
As a User, I want to measure pour data from a keg and store locally on the device, in order to capture data for batch analysis.

**Desc**
The device when monitoring a Keg is looking for changes in weight to signify a pour start and stop to calculate the pour size and timestamp. We've implemented a method for monitoring the keg for pours below. We want to store all readings locally ATC EEPROM not ESP EEPROM for batch export. Export will be a separate Story/Issue

**Method**
To measure keg pours we watch for stable weight readings in 5 second intervals. Every 5 seconds the weight is checked, if the weight is the same as the last reading the reading is considered stable. If the reading is different we know there is a pour occurring which may take more than one 5 second interval to complete so we begin a new comparison for a second stable reading. I've organized this using two setpoints in an array setpoint[0,1]. Once we have a second stable reading, we mesure the delta between the two points, reset setpointp[0] with the latest stable weight, and store the delta with timestamp as a pour to be transmitted to the system.

**Acceptance Criteria**

- The device follows the method to store a pour data point ie delta/timestamp
- The developer can execute a Serial.println() of the stored datapoints in EEPROM.

**Code Example**
```
int scaleSampleWeight(){
  scale.power_up();
  int sample = scale.get_units(25);
  //Throw away last bit
  sample = (sample/10)*10;
  scale.power_down();
  return sample;
}

void watch_keg(){
  int thisRead = scaleSampleWeight();
  
  Serial.println(thisRead);
  
  if(thisRead != setpoint[0]){
    
    if(thisRead == setpoint[1]){
      int delta = setpoint[1] - setpoint[0];
      
      if(abs(delta) > 10){
        trackpour(delta);
      }
      
      setpoint[0] = setpoint[1];
    
    }else{
      setpoint[1] = thisRead;
    }
  }
  
} 
```

**Comment by alexanderkjones** [comment]
@JLJTGR when's you're next availability for this issue?

**Comment by JLJTGR** [comment]
I'm still trying to figure out the usage of the HX711 library.  So far the only values I get out of it are 2147483647 and -2147483648 (0b01111111 11111111 11111111 11111111 and 0b10000000 00000000 00000000 00000000)

**Comment by alexanderkjones** [comment]
@JLJTGR  Here's working code for initializing HX711. I think you may have pins incorrect.
```
#include "HX711.h"

/* Initialize Scale*/
HX711 scale(12, 13);

void setup() {
  
  Serial.begin(115200);
  delay(100);

  scale.set_scale(43.5);
  //scale.tare();

}
```

**Comment by JLJTGR** [comment]
I don't think that's quite it at all.  Maybe the current library is fairly different than the one you were used to using.  14/12 are definitely the correct pins.  Your previous sample code assumes an int where the function returns a float.  You then decimate that truncated float by a power of ten, which causes any reading below 10 lbs to become zero.  Your scale calibration value is *vastly* different than my working one.

In short, I didn't understand what your code did, so when I used it blindly it didn't work and I didn't know how to fix it.  So I just did that portion from the ground up so I actually had an understanding of it.

**Comment by JLJTGR** [comment]
Output example...
```
Sleeping for 5 seconds...
2016-05-04 | 18:21:56
TrackPour: -12
Sleeping for 5 seconds...
2016-05-04 | 18:22:06
Sleeping for 5 seconds...
2016-05-04 | 18:22:16
Starting local AP: ESP_01560E...
Local AP IP: 192.168.4.1
Web server started...
Connecting to SSID: JLJTGR...
Connected; WAN IP: 192.168.173.100
Exporting 4 packets....
2016-05-04 18:19:57 = -14
2016-05-04 18:20:47 = -11
2016-05-04 18:21:17 = -11
2016-05-04 18:21:58 = -12
Done.
Disconnecting...
```

**Comment by JLJTGR** [comment]
I should mention that this code requires Arduino 1.6.8 to compile properly by the default IDE.  There is a bug that causes the ESP8266 libraries to crash the prototyping function of the default Arduino Builder and the compile will fail.
https://github.com/arduino/arduino-builder/issues/68
